### PR TITLE
parametric(cpp): enable prod and more parametric tests

### PR DESCRIPTION
**Pull request by dmehala**
## Motivation
dd-trace-cpp released a new version two days ago. This PR enable prod and enable more parametric tests.

## Changes
- CI
- Parametric tests for cpp

### CODE REVIEW

**Comment by cbeauchesne** [comment]
FYI #2506 is trying to fix C++ failures

File: tests/parametric/test_dynamic_configuration.py [hunk 1]
```diff
@@ -684,7 +685,8 @@ def test_trace_sampling_rules_override_rate(self, library_env, test_agent, test_
             }
         ],
     )
-    def test_trace_sampling_rules_with_tags(self, library_env, test_agent, test_library):
+    @bug(context.library == "cpp")
```
Comment (robertomonteromiguel) [inline tests/parametric/test_dynamic_configuration.py:688]: add the cause

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
There are test failures on CPP variant, but not in parametric. I guess it isn't related with this PR, right?

**Comment by dmehala** [comment]
> There are test failures on CPP variant, but not in parametric. I guess it isn't related with this PR, right?

No. If you are referencing #2506 , then it's appsec related.

File: tests/parametric/test_dynamic_configuration.py (hunk 1 above)
Comment (dmehala) [reply tests/parametric/test_dynamic_configuration.py:688]: > ~~add the cause~~
Could you add the cause? FTFY. Yes, I can.

Comment (dmehala) [inline tests/parametric/test_dynamic_configuration.py:688]:
```suggestion
    @bug(context.library == "cpp", reason = "JSON tag format in RC differs from the JSON tag format used in DD_TRACE_SAMPLING_RULES")
```

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**